analyses/tmb-compare-tcga/compare-tmb.Rmd

```diff
@@ -0,0 +1,276 @@
+---
+
+#### Purpose
+
+Plot the consensus tumor mutation burden statistics for PBTA in comparison to TCGA brain-related data. 
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/tmb-compare-tcga/compare-tmb.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+```{r}
+if (!("TCGAbiolinks" %in% installed.packages())) {
+  install.packages("TCGAbiolinks")
+}
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Load in these functions so we can use `maf_to_granges`
+source(file.path("..", "snv-callers", "util", "wrangle_functions.R"))
+```
+
+Declare names of input and output directories.
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+scratch_dir <- file.path("..", "..", "scratch")
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Create output directories for this analysis. 
+
+```{r}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+Custom function for plotting the TMB. 
+
+```{r}
+sina_plot <- function(data, colour = NULL, x_axis = "short_histology", title) {
+  # Given a data.frame with metadata and `tmb` information, make a sina plot from it
+  #
+  # Args:
+  #
+  # data : a data.frame with `tmb` as columns
+  # x_axis : A name of a column in data that you would like to make the x_axis from
+  #           Given as a character string. 
+  # colour: the color you would like the points to be
+  # title: The title for the plot. To be passed to `ggtitle`
+  #
+  # Returns:
+  #   A sina plot by histology group
+  data %>%
+    # eval parse bit will treat the variable as the text it has stored
+    ggplot2::ggplot(ggplot2::aes(x = reorder(eval(parse(text = x_axis)), tmb, mean), 
+                                 y = log10(tmb))) +
+    ggforce::geom_sina(color = colour, alpha = 0.3, maxwidth = 0.9) +
+    ggplot2::stat_summary(
+      fun.y = mean, fun.ymin = mean, fun.ymax = mean,
+      geom = "crossbar",
+      width = 0.95, size = 0.15
+    ) +
+    ggplot2::theme_classic() +
+    ggplot2::ylab("log10 (Number of Mutations per Mb)") +
+    ggplot2::xlab("") +
+    ggplot2::ylim(c(-3, 3)) +
+    ggplot2::theme(
+      axis.text.x = ggplot2::element_text(angle = 60, hjust = 1),
+      legend.position = "none"
+    ) +
+    ggplot2::ggtitle(title) + 
+    ggplot2::theme(
+    panel.grid.major.x = ggplot2::element_line(colour = c("gray93", "white"), size = 9)
+)
+}
+```
+
+## Set up consensus SNV PBTA data
+
+Download consensus SNV files. 
+
+```{r}
+# Declare file path for consensus file
+consensus_file <- file.path(data_dir, "snv-consensus_11122019", "consensus_mutation_tmb.tsv")
+
+# If this file hasn't been unzipped, unzip it.
+if (!file.exists(consensus_file)) {
+  # Get the files in this zip file
+  unzip(file.path(data_dir, "snv-consensus_11122019.zip"),
+    exdir = data_dir
+  )
+}
```

Comment:
Given how the download script is working now, I would take this out.

Reply:
Will do.